### DISCUSSION TITLE
Filtering

### DISCUSSION THREAD

**Discussion by nhorton** [ddnexus/pagy#437] [Q&A] [answered]
I have been using pagy a couple months and really appreciate it.

But I keep feeling like it should be handling filtering as well; is there any recommended way of doing filtering with Pagy, or something built in I missed?

**Comment by benkoshy** [comment]
> But I keep feeling like it should be handling filtering as well; is there any recommended way of doing filtering with Pagy, or something built in I missed

Pls elaborate on your requirements?

Typically filtering is handled by other means: ransack, or Meilisearch etc and ordinarily I would argue that it is probably outside the scope of pagination - but perhaps I misunderstood your use case?

**Comment by ddnexus** [comment]
@nhorton pagy tries to elude the Zawinski's Law:

“Every program attempts to expand until it can read mail."

🤭

**Comment by nhorton** [comment, marked as answer]
Honestly, I figured out 30 minutes after this post that the sorting
behavior I thought was in pagy was actually an addition someone had added
in our code. It is close enough to the basic filtering (not full search)
that a lot of the use cases need that it led me to the question.

Seeing that is not core pagy makes this irrelevant - apologies for the
misaligned question. Thanks all
